pkg/controller/accountclaim/reuse.go

```diff
@@ -223,12 +224,37 @@ func (r *ReconcileAccountClaim) cleanUpAwsAccountS3(reqLogger logr.Logger, awsCl
 			Bucket: aws.String(*bucket.Name),
 		}
 
+		DelError := fmt.Sprintf("Failed deleting S3 bucket: %s", *bucket.Name)
 		_, err = awsClient.DeleteBucket(&deleteBucketInput)
 		if err != nil {
-			delError := fmt.Sprintf("Failed deleting S3 bucket: %s", *bucket.Name)
-			reqLogger.Error(err, delError)
-			awsErrors <- delError
-			return err
+			if aerr, ok := err.(awserr.Error); ok {
+				DelObjErr := fmt.Sprintf("Failed deleting items in S3 bucket: %s", *bucket.Name)
+				switch aerr.Code() {
+				// if bucket is not empty empty the bucket and attempt to delete again
+				case "BucketNotEmpty":
```

Comment:
Do you think this logic should be part of the error? I think the correct logic would be to check if the bucket is empty or not and then perform the appropriate action.

Reply:
I had that thought @jharrington22  but so far we only hit this issue with the velero buckets and then we delete it. We can check if the bucket is empty first , but I see more logic needed and logic that may not necessarily need to be run every time, but I am happy to change it if you guys want, but it would probably have to be tomorrow at that point.